test/named-parameter-order-test.js

```diff
@@ -0,0 +1,36 @@
+var shuffle = require('shuffle-array');
+var test = require('tap').test;
+var docuri = require('..');
+
+var simulateWrongOrderObjectKeys = function(cb){
+  var orig = Object.keys;
+  Object.keys = function(obj){
+    var keys = orig(obj);
+    var real = keys.toString();
+    while (real === keys.toString())
+      shuffle(keys);
```

Comment:
We do not need to test for shuffle order, just make sure that the order does not match insertion order.
I have merged it for now, it can be changed later.